### PR TITLE
map stop_gradient over data structure. otherwise it is silently a no-op

### PR DESCRIPTION
**Pull request by joschu**
Previous behavior:
```
jax.grad(lambda x: jax.lax.stop_gradient({'foo':x})['foo'])(3.) # returns array(1.0)
```
After the change, it returns `array(0.0)`.

It's useful to apply a network or sub-network but stop gradient on the parameters.

### CODE REVIEW

**Review by mattjj — APPROVED** [review]
Makes sense, thanks! Will merge when Travis finishes.

(Travis is taking almost 40 minutes nowadays... as a temporary solution we'll reduce the number of test cases sampled per method from 100 to 50, and as a longer-term solution we'll probably invest in our own CI infrastructure.)
